driver-core/src/main/java/com/datastax/driver/core/AbstractAddressableByIndexData.java

```diff
@@ -256,22 +209,14 @@ public T setInet(int i, InetAddress v) {
     }
 
     public T setUDTValue(int i, UDTValue v) {
-        checkType(i, UDT);
         return setValue(i, codecFor(i, UDTValue.class).serialize(v, protocolVersion));
     }
 
     public T setTupleValue(int i, TupleValue v) {
-        checkType(i, TUPLE);
         return setValue(i, codecFor(i, TupleValue.class).serialize(v, protocolVersion));
     }
 
     @Override
-    public <V> T setObject(int i, V v) {
```

Comment:
Any concerns about a loss of utility here?   I think depending on class/typetoken/typecodec in set() is likely a good move in that the user needs to specify what targetClass they intend to be communicating here.  I do see someone wanting the use case of "Here's an object, I have no idea what this is, but let the driver figure out what to do with it", but that on its face sounds like it could have unreliable/unpredictable results.